### PR TITLE
Add google tag manager datalayer pushes to the application form.

### PR DESCRIPTION
**Pull request by douglaschan32167**
## Proposed changes
Log when the application is viewed, when they advance a page, and when they submit. Also refactor the subscribe to newsletter dataLayer push. Tested locally.

## Types of changes

- [x] I've labeled this PR with appropriate labels (`comp/*` component, bug/feature, documentation and other less important ones).

## Checklist
_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging your code._

- [x] I've linked the proper Github Issue(s) to my PR
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests (unit, integration...etc.) that prove my fix is effective or that my change works
- [x] I have added or updated necessary documentation (if appropriate). You can search, for instance using the [`silver searcher`](https://github.com/ggreer/the_silver_searcher) `ag <STRING>` to find all usages in the repository.
- [x] I have notified the team of any breaking or important changes

### CODE REVIEW

File: frontend/src/components/Subscribe/Subscribe.jsx [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 import React, { useState } from 'react';
-import TagManager from 'react-gtm-module';
 import { useTranslation } from 'react-i18next';
 import { Button, Input } from 'semantic-ui-react';
 import styled from 'styled-components';
 import mixins from 'styles/mixins';
 import theme from 'styles/theme';
 import { SubscribeNewsletter } from 'utils/agent';
+import TagManager from 'react-gtm-module';
+import {logSubscribe} from "../../utils/google_tag_manager_helpers";
```
Comment (iksoosong) [inline frontend/src/components/Subscribe/Subscribe.jsx:10]: nit: make this absolute and not relative
`utils/...` works

File: frontend/src/pages/ApplicationForm/ApplicationForm.jsx [hunk 2]
```diff
@@ -34,6 +34,11 @@ import {
 
 import { useTranslation } from 'react-i18next';
 import PageContainer from 'layout/PageContainer';
+import {
+  logApplicationCompletion,
+  logApplicationProgress,
+  logApplicationView
+} from "../../utils/google_tag_manager_helpers";
```
Comment (iksoosong) [inline frontend/src/pages/ApplicationForm/ApplicationForm.jsx:41]: same here on the import

File: frontend/src/utils/google_tag_manager_helpers.js [hunk 3]
```diff
@@ -1,10 +1,62 @@
...
   const pageViewTagManagerArgs = {
     dataLayer: {
       'event': 'application_click'
     }
   }
   TagManager.dataLayer(pageViewTagManagerArgs)
+}
+
+export const logApplicationView = () => {
+  console.log("in log application view");
```
Comment (iksoosong) [inline frontend/src/utils/google_tag_manager_helpers.js:31]: remove log

Comment (douglaschan32167) [reply frontend/src/utils/google_tag_manager_helpers.js]: done

File: frontend/src/pages/ApplicationForm/ApplicationForm.jsx (hunk 2 above)
Comment (douglaschan32167) [reply frontend/src/pages/ApplicationForm/ApplicationForm.jsx]: done

File: frontend/src/components/Subscribe/Subscribe.jsx (hunk 1 above)
Comment (douglaschan32167) [reply frontend/src/components/Subscribe/Subscribe.jsx]: done

**Pull request merged**